HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1399,77 +1425,241 @@ def self.add_ee_for_vent_fan_power(model, obj_name, frac_lost, is_cfis, pow = 0.
-        living_ach50 = calc_air_leakage_at_diff_pressure(0.65, living_ach, air_infil.house_pressure, 50.0)
-      elsif air_infil.unit_of_measure == HPXML::UnitsACHNatural
-        if @apply_ashrae140_assumptions
-          living_const_ach = air_infil.air_leakage
+  def self.setup_mech_vent_vars_actuators(model:, program:)
+    # Actuators for mech vent fan
+    sens_name = "#{Constants.ObjectNameMechanicalVentilationHouseFan} sensible load"
+    fan_sens_load_actuator = create_sens_lat_load_actuator_and_equipment(model: model, name: sens_name, space: @living_space, frac_lat: 0.0, frac_lost: 0.0)
+    lat_name = "#{Constants.ObjectNameMechanicalVentilationHouseFan} latent load"
+    fan_lat_load_actuator = create_sens_lat_load_actuator_and_equipment(model: model, name: lat_name, space: @living_space, frac_lat: 1.0, frac_lost: 0.0)
+    program.addLine("Set #{fan_sens_load_actuator.name} = 0.0")
+    program.addLine("Set #{fan_lat_load_actuator.name} = 0.0")
+    # QWHV to track implemented whole house ventilation rate
+    program.addLine('Set QWHV = 0.0')
+    # Air property at inlet nodes on both sides
+    program.addLine("Set OASupInPb = #{@pbar_sensor.name}") # oa barometric pressure
+    program.addLine("Set OASupInTemp = #{@tout_sensor.name}") # oa db temperature
+    program.addLine("Set OASupInW = #{@wout_sensor.name}") # oa humidity ratio
+    program.addLine('Set OASupRho = (@RhoAirFnPbTdbW OASupInPb OASupInTemp OASupInW)')
+    program.addLine('Set OASupCp = (@CpAirFnW OASupInW)')
+    program.addLine('Set OASupInEnth = (@HFnTdbW OASupInTemp OASupInW)')
+
+    program.addLine("Set ZoneTemp = #{@tin_sensor.name}") # zone air temperature
+    program.addLine("Set ZoneW = #{@win_sensor.name}") # zone air humidity ratio
+    program.addLine('Set ZoneCp = (@CpAirFnW ZoneW)')
+    program.addLine('Set ZoneAirEnth = (@HFnTdbW ZoneTemp ZoneW)')
+
+    return program, fan_sens_load_actuator, fan_lat_load_actuator
+  end
+
+  def self.apply_erv_hrv_fans(model:, program:, vent_mech_erv_hrv:, hrv_erv_effectiveness_map:, fan_sens_load_actuator: nil, fan_lat_load_actuator: nil, preconditioned: true)
+    # Apply ERV HRV calculations for both infiltration program and shared system preconditioning program
+    # Use preconditioned label to specify if it's called by preconditioning program, if so, disable the load actuators.
+    if not vent_mech_erv_hrv.empty?
+      # ERV/HRV EMS load model
+      # E+ ERV model is using standard density for MFR calculation, caused discrepancy with other system types.
+      # Therefore ERV is modeled within EMS infiltration program
+      # combine ERV/HRV to a single system with weighted effectivenesses
+      vent_mech_sens_eff = 0.0
+      vent_mech_lat_eff = 0.0
+      oa_flow_rate_sum = vent_mech_erv_hrv.map { |fan| fan.average_oa_flow_rate }.sum
+      tot_flow_rate_sum = vent_mech_erv_hrv.map { |fan| fan.average_unit_flow_rate }.sum
+      vent_mech_erv_hrv.each do |vent_fan|
```

Comment:
The ERV/HRV method here did two things: 1. combine ERV/HRV effectiveness to a single one. 2. call the apply_fan_load_to_infil_program for load calculations.